### DISCUSSION TITLE
usb audio input

### DISCUSSION THREAD

**Discussion by GautierRtlc** [qqq89513/stm32f7-usb-vu-meter#11] [Q&A]
Hello, I hope you are well.

I am currently working (as a student) on a stm32f7 board and would like to have a mic input via USB. I'm working on windows10 and using the STM32CubeIDE.

I searched the web and saw your project. I will not use the LCD part. The goal is to have like a sound card with a mic input and a headphone output. Do you think your project can be easily adapted to do what I want?

Best regards, 
Gautier

**Comment by qqq89513** [comment]
This project includes overhead of GUI framework (TouchGFX).
I'm not sure if you can remove the GUI completely since STM32CubeMX has many bugs.
And there is nothing about USB device as mic either.

I think what this project can help you is the sound output.
- [usbd_audio_if.c](https://github.com/qqq89513/stm32f7-usb-audio-visualizer/blob/master/USB_DEVICE/App/usbd_audio_if.c) defines what to do with USB speaker device
- [stm32746g_discovery_audio.c](https://github.com/qqq89513/stm32f7-usb-audio-visualizer/blob/master/Drivers/BSP/STM32746G-Discovery/stm32746g_discovery_audio.c) is the driver to the onboard DAC (output through 3.5mm jack).

[This project](http://www.tjaekel.com/DiscoveryF7Audio/index.html) also shows how to manipulate onboard DAC and mic.

It's a good idea that you copy files from this project to the one you created, since this one might be highly coupled with TouchGFX.

**Reply by GautierRtlc** [reply]
Thank you very much for your quick and precise answer.

I will be able to look into what you have told me in more detail. Especially to manage the audio output.

Thanks again!

**Comment by GautierRtlc** [comment]
Hello, I have another question^^

Do you think it's possible to have access to an audio buffer in order to apply audio filters on a signal we control?
I saw that we use a pointer to a buffer in the "BSP_AUDIO_OUT_Play" function.

Gautier,
